tests/testthat/test-utils.R

```diff
@@ -153,3 +153,17 @@ test_that("testing_port", {
     }
   )
 })
+
+test_that("get_dbname() works", {
+  expect_equal(get_dbname(list(dbname = "db")), "db")
+  expect_equal(get_dbname(list(Database = "db")), "db")
+  expect_equal(get_dbname(list("db")), "db")
+  expect_equal(get_dbname(list("db1", dbname = "db2")), "db2")
+  expect_equal(get_dbname(list("db1", Database = "db2")), "db2")
+  # ODBC (shouldn't be an error)
+  expect_error(get_dbname(list(dsn = "dsn")))
```

Comment:
Oops, I totally missed this when I wrote this function. I can definitely add in a case that uses the `dsn` if it's there (just like we already check `dbname` and `Database`). My first thought was that ideally this would use the `database` field in the data source itself, but I'm not seeing an easy way of doing that (`odbcListDataSources()` gives the name + description only). If I'm missing something in {odbc} that would allow that I would appreciate the pointer.

Though thinking about this a bit more, likely the `dsn` is what one would want regardless since guaranteeing that the data source config files are present on the testing system would be a pain.